app/app-e2e-tests/e2e/example.spec.ts

```diff
@@ -0,0 +1,135 @@
+// electronApp.process().stdout.on('data', (data) => console.log(`stdout: ${data}`));
+// electronApp.process().stderr.on('data', (error) => console.log(`stderr: ${error}`));
+
+// test('launch app', async () => {
+//   const electronApp = await _electron.launch({ cwd: '/home/vtaylor/headlamp/app/electron', args: ['main.js'] })
+//   // close app
+//   await electronApp.close()
+// })
+
+// Function to run a shell command and return a promise
+async function runCommand(command: string, args: string[] = [], options = {}): Promise<void> {
+  return new Promise((resolve, reject) => {
+    const process = spawn(command, args, { ...options, detached: true, stdio: 'ignore' });
+
+    process.on('error', error => {
+      console.error(`Error executing command: ${command}\n${error}`);
+      reject(error);
+    });
+
+    process.on('close', code => {
+      if (code !== 0) {
+        console.error(`Command failed with exit code ${code}`);
+        reject(new Error(`Command failed with exit code ${code}`));
+      } else {
+        resolve();
+      }
+    });
+
+    // Detach the process and resolve immediately
+    process.unref();
+    resolve();
+  });
+}
+
+test('launch app', async () => {
+  // Launch Electron app.
+  const electronApp = await electron.launch({
+    cwd: '/home/vtaylor/headlamp/app/electron',
+    timeout: 0,
+    args: ['main.js'],
+  });
+  electronApp.process().stdout.on('data', data => console.log(`stdoutX: ${data}`));
+  electronApp.process().stderr.on('data', error => console.log(`stderrX: ${error}`));
+  // Evaluation expression in the Electron context.
+  // const appPath = await electronApp.evaluate(async (electron) => {
+  //   // This runs in the main Electron process, |electron| parameter
+  //   // here is always the result of the require('electron') in the main
+  //   // app script.
+  //   // return electron.getAppPath();
+  // });
+
+  // Get the first window that the app opens, wait if necessary.
+  const window = await electronApp.firstWindow();
+  // await window.goto('https://playwright.dev/');
+  await window.waitForLoadState('domcontentloaded');
+
+  // await window.goto('https://playwright.dev/');
+  // await window.getByRole('link', { name: 'Get started' }).click();
+
+  await window.goto('http://localhost:3000/');
+  await window.getByRole('button', { name: 'Home' }).click();
+
+  // await electronApp.browserWindow(page);
+  // Print the title.
+  // console.log(await window.title());
+  // Capture a screenshot.
+  // await window.screenshot({ path: 'intro.png' });
+
+  // await headlampPage.authenticate();
+  // await headlampPage.hasNetworkTab();
+  // Direct Electron console to Node terminal.
+  window.on('console', console.log);
+  // Click button.
+  // await window.click('text=Click me');
+});
+
+test('launch app with integrated shell commands', async () => {
+  // Start the frontend server
+  await runCommand('npm', ['start'], { cwd: '/home/skoeva/headlamp/frontend' });
```

Comment:
The automation needs a way to be closed after running the commands for npm start etc, this could be done by running this all before testing using beforeAll

The server being ran this also has issues as it does not auto run to the next lines after launching, needing manual input for either continue or quit as it clashes

Reply:
I think we may need to have this be used offline like the current base project for in cluster and web mode